core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -86,8 +86,17 @@ private[spark] class EventLoggingListener(
    * Creates the log file in the configured log directory.
    */
   def start() {
-    if (!fileSystem.isDirectory(new Path(logBaseDir))) {
-      throw new IllegalArgumentException(s"Log directory $logBaseDir does not exist.")
+    try {
+      if (!fileSystem.getFileStatus(new Path(logBaseDir)).isDir()) {
+        throw new IllegalArgumentException(s"Log directory $logBaseDir is not a directory.")
+      }
+    } catch {
+      case ioe: FileNotFoundException => {
+        logWarning(s"Log directory $logBaseDir does not exist. Creating with default file " +
+          "permissions...")
+        new File(logBaseDir).mkdir()
```

Comment:
We should remove this line, since we need to use `fileSystem` to create the file.  Also, I think we should use `mkdir` and not `mkdirs`, since I think that the old code failed if they didn't exist.